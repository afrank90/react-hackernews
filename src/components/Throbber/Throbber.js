import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Throbber = ({ className }) => (
  <div className="c-throbber">
    <div className={className} />
  </div>
);

Throbber.defaultProps = {
  className: ''
};

Throbber.propTypes = {
  className: PropTypes.string
};

export default Throbber;
