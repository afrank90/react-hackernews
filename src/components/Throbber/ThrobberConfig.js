export default
{
  'size_small': 'small',
  'size_medium': 'medium',
  'size_large': 'large',
};