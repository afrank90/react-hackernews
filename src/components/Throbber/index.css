.c-throbber {
  display: flex;
  justify-content: center;
}

.c-throbber__small,
.c-throbber__medium,
.c-throbber__large {
  width: 100%;
  height: 0;
  padding: 0;
  background-image: url(./Assets/throbber.svg);
  background-size: cover;
}

.c-throbber__small {
  width: 40px;
  padding-bottom: 40px;
}

.c-throbber__medium {
  width: 60px;
  padding-bottom: 60px;
}

.c-throbber__large {
  width: 80px;
  padding-bottom: 80px;
}
