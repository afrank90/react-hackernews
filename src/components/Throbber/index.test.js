import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Throbber from './';

Enzyme.configure({ adapter: new Adapter() });

describe('Throbber', () => {
  const props = {
    className: 'test-class-name'
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Throbber {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has valid snapshot', () => {
    const component = renderer.create(<Throbber {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
