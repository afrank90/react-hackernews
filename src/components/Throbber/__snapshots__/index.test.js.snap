// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Throbber has valid snapshot 1`] = `
<div
  className="c-throbber"
>
  <div
    className="test-class-name"
  />
</div>
`;
